src/Interpreters/Aggregator.cpp

```diff
@@ -588,7 +588,9 @@ Aggregator::Aggregator(const Block & header_, const Params & params_)
     : header(header_)
     , keys_positions(calculateKeysPositions(header, params_))
     , params(params_)
-    , tmp_data(params.tmp_data_scope ? std::make_unique<TemporaryDataOnDisk>(params.tmp_data_scope, CurrentMetrics::TemporaryFilesForAggregation) : nullptr)
+    , tmp_data(params.tmp_data_scope ? std::make_unique<TemporaryDataOnDisk>(params.tmp_data_scope,
+        TemporaryDataOnDiskSettings{.compression_codec = params.temporary_data_compression_codec},
```

Comment:
Can't we set codec once when `TemporaryDataOnDiskScope` for query is created from query settings and for global (root) scope from server settings, not to pass the value of the setting to each place that uses `TemporaryDataOnDisk` ?

https://github.com/ClickHouse/ClickHouse/blob/d045ab150ed5659f143beca5e50d0b72dae3bf78/src/Interpreters/ProcessList.cpp#L211-L212

https://github.com/ClickHouse/ClickHouse/blob/d045ab150ed5659f143beca5e50d0b72dae3bf78/src/Interpreters/Context.cpp#L1143

Reply:
If in `ProcessList` https://github.com/ClickHouse/ClickHouse/blob/d045ab150ed5659f143beca5e50d0b72dae3bf78/src/Interpreters/ProcessList.cpp#L211 we do not have `user_process_list.user_temp_data_on_disk`, that means user cannot use temporary data right ?

Reply:
Global (root) temp data scope will be used then, I suppose.

Reply:
Updated.